actionpack/lib/action_dispatch/http/cache.rb

```diff
@@ -167,9 +167,7 @@ def prepare_cache_control!
         end
 
         def handle_conditional_get!
-          if etag? || last_modified? || !@cache_control.empty?
```

Comment:
I tracked down the original intent of this logic in https://github.com/rails/rails/blob/503ce1d01ce6c8eee9818f4e76a9f880bb1a291d/actionpack/lib/action_dispatch/http/response.rb#L199-L214. It looks like the intent was to always call `set_conditional_cache_control!`..except in the case where we default to a `no-cache` policy. It looks like the other cases that ended up calling!` were removed in https://github.com/rails/rails/commit/74dd8a3681c6984ea35c879f88c6a87521b58ec2#diff-b417aa8ac6697cf444c7135a09f20536L91.